tests/check_framework/test_model_checks.py

```diff
@@ -403,6 +403,18 @@ class Child(Parent):
 
         self.assertEqual(checks.run_checks(app_configs=self.apps.get_app_configs()), [])
 
+    def test_explicit_pk_with_abstract_parent(self):
+        class Abstract(models.Model):
+            class Meta:
+                abstract = True
+
+        class Child(Abstract):
+            id = models.AutoField(primary_key=True)
+
+        self.assertEqual(checks.run_checks(app_configs=self.apps.get_app_configs()), [])
```

Comment:
This bit wasn't broken before, right? So, there should be already tests covering this behavior.

Reply:
Well, it was intended to test that child of an abstract parent also works fine. :)
But if it does not make sense in this context, we can drop it, right.

Reply:
I would keep the tests as atomic as possible. That aside, if you want to test the behavior for `abstract=False`, I would recommend dropping the PK and test that the check fails, hence the code inside the if-statement is actually being executed.

Reply:
@codingjoe very good point, Johannes :) I added a separate test to cover that check works in this case.